services/graph/pkg/service/v0/sharedwithme.go

```diff
@@ -38,185 +40,251 @@ func (g Graph) listSharedWithMe(ctx context.Context) ([]libregraph.DriveItem, er
+			// return no remoteItem
+			fallthrough
+		case collaboration.ShareState_SHARE_STATE_REJECTED:
+			// what to return here? same as pending?
+			statRequest.Ref = &storageprovider.Reference{
+				ResourceId: receivedShare.GetShare().GetResourceId(),
+			}
+		default:
 			continue
 		}
 
-		driveItem := &libregraph.DriveItem{}
-
-		statResponse, err := gatewayClient.Stat(ctx, &storageprovider.StatRequest{Ref: &storageprovider.Reference{ResourceId: share.GetResourceId()}})
-		if err != nil {
+		statResponse, err := gatewayClient.Stat(ctx, statRequest)
+		if errCode := errorcode.FromCS3Status(statResponse.GetStatus(), err); errCode != nil {
 			g.logger.Error().Err(err).Msg("could not stat")
 			continue
 		}
-		if statResponse.GetStatus().GetCode() != rpc.Code_CODE_OK {
-			g.logger.Error().Err(err).Msg("invalid stat response")
-			continue
-		}
-		resourceInfo := statResponse.GetInfo()
 
-		var driveOwner *libregraph.Identity
+		var commonResourceOwner *libregraph.Identity
 		if userID := statResponse.GetInfo().GetOwner(); userID != nil {
 			if user, err := g.identityCache.GetUser(ctx, userID.GetOpaqueId()); err != nil {
 				g.logger.Error().Err(err).Msg("could not get user")
 				continue
 			} else {
-				driveOwner = &libregraph.Identity{
+				commonResourceOwner = &libregraph.Identity{
 					DisplayName: user.GetDisplayName(),
 					Id:          libregraph.PtrString(user.GetId()),
 				}
 			}
 		}
 
-		var shareCreator *libregraph.Identity
-		if userID := share.GetCreator(); userID != nil {
+		var commonShareCreator *libregraph.Identity
+		if userID := receivedShare.GetShare().GetCreator(); userID != nil {
 			if user, err := g.identityCache.GetUser(ctx, userID.GetOpaqueId()); err != nil {
 				g.logger.Error().Err(err).Msg("could not get user")
 				continue
 			} else {
-				shareCreator = &libregraph.Identity{
+				commonShareCreator = &libregraph.Identity{
 					DisplayName: user.GetDisplayName(),
 					Id:          libregraph.PtrString(user.GetId()),
 				}
 			}
 		}
 
-		if cTime := share.GetCtime(); cTime != nil {
-			driveItem.CreatedDateTime = libregraph.PtrTime(cs3TimestampToTime(cTime))
-		}
-
-		driveItem.ETag = libregraph.PtrString(strings.Trim(statResponse.GetInfo().GetEtag(), "\""))
+		var commonPermission *libregraph.Permission
+		{
+			permission := libregraph.NewPermission()
 
-		if id := share.GetId().GetOpaqueId(); id != "" {
-			driveItem.Id = libregraph.PtrString(id)
-		}
+			if id := receivedShare.GetShare().GetId().GetOpaqueId(); id != "" {
+				permission.SetId(id)
+			}
 
-		if mTime := share.GetMtime(); mTime != nil {
-			driveItem.LastModifiedDateTime = libregraph.PtrTime(cs3TimestampToTime(mTime))
-		}
+			if permissionSet := statResponse.GetInfo().GetPermissionSet(); permissionSet != nil {
+				if actions := unifiedrole.CS3ResourcePermissionsToLibregraphActions(*permissionSet); len(actions) > 0 {
+					permission.SetLibreGraphPermissionsActions(actions)
+				}
```

Comment:
the `@libre.graph.permissions.actions` property should only be set when we can't resolve the `permissionSet` to a  `unifiedRole`. So I think this belongs to the `else` branch of the below `if` statement.

Reply:
done